pkg/client/unversioned/clientcmd/client_config.go

```diff
@@ -358,21 +358,21 @@ func (inClusterClientConfig) ClientConfig() (*restclient.Config, error) {
 	return restclient.InClusterConfig()
 }
 
-func (inClusterClientConfig) Namespace() (string, error) {
```

Comment:
The interface is correct - this implementation just happens not to return an error.  The bool is "was this specified" by the user.  The string is the value.  We use the bool all over the code to determine between user passed "--namespace" on the CLI vs getting the "default" namespace.